### PR TITLE
Sea 178 test actions loading

### PR DESCRIPTION
**Pull request by jhazentia**
- Updated precommit versions and autoformatted
- added ability to set calibration adjust for data product action (useful for testing with mock)
- sensor cal only added to metadata and measurement result if calibration adjust is true
- added plugin_name property to sigan_iface; this is used to put the plugin name in the status page and in the metadata
- remove "calibration_datetime" in global section of metadata, replaced by calibration object in capture
- fix calculation of duration in metadata in the m4s action - this appeared to not be converted to milliseconds as expected (always showed 0 in metadata)
- changed the logic in the base action configure_preselector method to first check if preselector is configurable
- test data product action updated to be consistent with cbrs measure baseline action
- actions and test actions (except logger) only loaded if SIGAN_MODULE and SIGAN_CLASS are set to the mock sigan in scos-actions
- ~added sigan to mock GPS constructor just like the scos-usrp USRPLocation class~
- Due to sigan and GPS objects being initialized in scos-sensor instead of scos-usrp, sensor object now passed to GPS methods in order for scos-usrp to access USRP API for GPS. Previously sigan was passed to GPS Interface implementation through constructor.
- Updated to be compatible with the latest scos-actions:
  - Acquire IQ called though sensor object rather than sigan with sensor object now handling calibration adjustments
  - Sensor overload code moved from scos-usrp to sensor in scos-actions
  - remove clock_rate_lookup_by_sample_rate related code, instead scos-usrp will always maximize clock_rate based on sample rate
- Check if temperature exists in cal data before adding to cal metadata

### CODE REVIEW

File: scos_actions/actions/acquire_sea_data_product.py [hunk 1]
```diff
@@ -630,14 +632,20 @@ def capture_iq(self, params: dict) -> dict:
         nskip = utils.get_parameter(NUM_SKIP, params)
         num_samples = int(params[SAMPLE_RATE] * duration_ms * 1e-3)
         # Collect IQ data
+        self.cal_adjust = True
+        for key, value in params.items():
+            logger.debug(f"param {key}={value}")
+        self.cal_adjust = utils.get_parameter(CAL_ADJUST, params)
```
Comment (dboulware) [inline scos_actions/actions/acquire_sea_data_product.py:638]: Might want to either check if the CAL_ADJUST param exists or catch the exception if it doesn't and set it to true.

File: scos_actions/hardware/mocks/mock_gps.py [hunk 2]
```diff
@@ -7,6 +7,9 @@
 
 
 class MockGPS(GPSInterface):
+    def __init__(self, sigan):
```
Comment (dboulware) [inline scos_actions/hardware/mocks/mock_gps.py:10]: Why does this need a sigan?

File: scos_actions/configs/actions/test_SEA_CBRS_Measure_Baseline.yml [hunk 3]
```diff
@@ -1,26 +1,22 @@
 nasctn_sea_data_product:
-  name: test_nasctn_sea_data_product
+  name: test_SEA_CBRS_Measure_Baseline
   rf_path: antenna
```
Comment (dboulware) [inline scos_actions/configs/actions/test_SEA_CBRS_Measure_Baseline.yml:3]: I think this same rf_path: antenna param should be added to all of the test actions to allow them to work with a configurable preselector or without a preselector. Currently, they will all fail on a sensor if it has a configurable preselector.

File: scos_actions/hardware/mocks/mock_gps.py (hunk 2 above)
Comment (jhazentia) [reply scos_actions/hardware/mocks/mock_gps.py]: Fixed in 91e3eff

File: scos_actions/actions/acquire_sea_data_product.py (hunk 1 above)
Comment (jhazentia) [reply scos_actions/actions/acquire_sea_data_product.py]: Fixed in d06098f

File: scos_actions/configs/actions/test_SEA_CBRS_Measure_Baseline.yml (hunk 3 above)
Comment (jhazentia) [reply scos_actions/configs/actions/test_SEA_CBRS_Measure_Baseline.yml:3]: Fixed in 8d205e5

File: scos_actions/hardware/sensor.py [hunk 4]
```diff
@@ -246,6 +246,26 @@ def recompute_calibration_data(self, params: dict) -> None:
         if not recomputed:
             logger.warning("Failed to recompute calibration data")
 
+    def check_sensor_overload(self, data) -> bool:
```
Comment (dboulware) [inline scos_actions/hardware/sensor.py:249]: It looks like data is expected to be an ndarray? I suggest adding a type hint. Also, is there any danger that data could not be cast to np.complex64? I'm not sure.

File: scos_actions/hardware/sensor.py [hunk 5]
```diff
@@ -246,6 +246,26 @@ def recompute_calibration_data(self, params: dict) -> None:
...
+        """Check for sensor overload in the measurement data."""
+        measured_data = data.astype(np.complex64)
+
+        time_domain_avg_power = 10 * np.log10(np.mean(np.abs(measured_data) ** 2))
+        time_domain_avg_power += (
+            10 * np.log10(1 / (2 * 50)) + 30
+        )  # Convert log(V^2) to dBm
+        # explicitly check is not None since 1db compression could be 0
+        if self.sensor_calibration_data["compression_point"] is not None:
```
Comment (dboulware) [inline scos_actions/hardware/sensor.py:258]: Suggest moving this above so you don't compute the avg power if you don't know the compression point.

Comment (dboulware) [reply scos_actions/hardware/sensor.py:258]: Looks like this is only called if there is compression_point so maybe the previous comment can be disregarded unless there is a chance check_sensor_overload could be called in a different flow.

File: scos_actions/discover/__init__.py [hunk 6]
```diff
@@ -3,18 +3,10 @@
...
-    "logger": Logger(),
-}
-test_actions = {
-    "test_sync_gps": SyncGps(parameters={"name": "test_sync_gps"}),
-    "test_monitor_sigan": MonitorSignalAnalyzer(
-        parameters={"name": "test_monitor_sigan"}
-    ),
-    "logger": Logger(),
-}
+actions = {"logger": Logger()}
```
Comment (dboulware) [inline scos_actions/discover/__init__.py:8]: Should we remove logger from the actions? Not a big deal either way.

**Review by dboulware — CHANGES_REQUESTED** [review]
Looks good. See comments for a couple minor suggestions.

**Pull request merged**